uber/config.py

```diff
@@ -698,27 +699,67 @@ def ADMIN_WRITE_ACCESS_SET(self):
         }
+        
+    @request_cached_property
+    def SITE_MAP(self):
+        public_site_sections, public_pages, pages = self.gettable_site_sections_and_pages
+        
+        accessible_site_sections = {section: pages for section, pages in pages.items() 
+                                    if c.has_section_or_page_access(page_path=section, include_read_only=True)}
+        for section in accessible_site_sections:
+            accessible_site_sections[section] = [page for page in accessible_site_sections[section] 
+                                                 if c.has_section_or_page_access(page_path=page['path'], include_read_only=True)]
+            
+        return sorted(accessible_site_sections.items())
+    
+    @cached_property
+    def gettable_site_sections_and_pages(self):
+        """
+        Introspects all available pages in the application and returns several data structures for use in displaying them.
+        Returns:
+            public_site_sections (list): a list of site sections that are accessible to the public, e.g., 'preregistration'
+            public_pages (list): a list of individual pages in non-public site sections that are accessible to the public,
+                prepended by their site section; e.g., 'registration_register' for registration/register
+            pages (defaultdict(list)): a dictionary with keys corresponding to site sections, each key containing a list
+                of key/value pairs for each page inside that section.
+                Example: 
+                    pages['registration'] = [
+                        {'name': 'Arbitrary Charge Form', 'path': '/registration/arbitrary_charge_form'},
+                        {'name': 'Comments', 'path': '/registration/comments'},
+                        {'name': 'Discount', 'path': '/registration/discount'},
+                    ]
+        """
+        public_site_sections = ['static_views', 'angular', 'public', 'staffing']
+        public_pages = []
+        site_sections = cherrypy.tree.apps[c.CHERRYPY_MOUNT_PATH].root
+        modules = {name: getattr(site_sections, name) for name in dir(site_sections) if not name.startswith('_')}
+        pages = defaultdict(list)
+        for module_name, module_root in modules.items():
+            page_method = getattr(site_sections, module_name)
+            if getattr(page_method, 'public', False):
+                public_site_sections.append(module_name)
+            for name in dir(module_root):
+                method = getattr(module_root, name)
+                if getattr(page_method, 'public', False):
+                            public_pages.append(module_name + "_" + name)
+                if getattr(method, 'exposed', False):
+                    spec = inspect.getfullargspec(unwrap(method))
+                    has_defaults = len([arg for arg in spec.args[1:] if arg != 'session']) == len(spec.defaults or [])
```

Comment:
I realize that this line is copied over from elsewhere and thus isn't new to this PR, but we should consider whether we'd also like to include ``out`` as an exception here, for ``@csv`` exports, which if memory serves don't show up on the site map, i.e.

```python
if arg not in ['session', 'out']
```

Unless maybe we already updated the ``@csv`` decorator to set ``site_mappable`` or something, in which case [ignore me](https://www.youtube.com/watch?v=FMNJuSl91qY&t=7s).

Reply:
We do already have CSV files set to site mappable if they don't have extra parameters, so this should be fine as-is.